packages/ui-text-area/src/TextArea/props.ts

```diff
@@ -39,110 +39,110 @@ import { TextareaHTMLAttributes } from 'react'
 type TextAreaOwnProps = {
   label: React.ReactNode
   id?: string
-  size?: 'small' | 'medium' | 'large'
-  layout?: 'stacked' | 'inline'
-  autoGrow?: boolean
-  resize?: 'none' | 'both' | 'horizontal' | 'vertical'
-  width?: string
-  height?: string
-  maxHeight?: number | string
-  messages?: FormMessage[]
-  inline?: boolean
-  placeholder?: string
-  disabled?: boolean
-  readOnly?: boolean
-  required?: boolean
-  textareaRef?: (...args: any[]) => any
-  defaultValue?: string
-  value?: any // TODO: controllable(PropTypes.string)
-  onChange?: (...args: any[]) => any
-}
-
-type PropKeys = keyof TextAreaOwnProps
-
-type AllowedPropKeys = Readonly<Array<PropKeys>>
-
-type TextAreaProps = TextAreaOwnProps &
-  WithStyleProps<TextAreaTheme, TextAreaStyle> &
-  OtherHTMLAttributes<
-    TextAreaOwnProps,
-    TextareaHTMLAttributes<TextAreaOwnProps>
-  >
-
-type TextAreaStyle = ComponentStyle<
-  'textArea' | 'textAreaLayout' | 'textAreaOutline'
->
-
-const propTypes: PropValidators<PropKeys> = {
-  label: PropTypes.node.isRequired,
-  id: PropTypes.string,
   /**
    * sets the font-size for the textarea
    */
-  size: PropTypes.oneOf(['small', 'medium', 'large']),
-  layout: PropTypes.oneOf(['stacked', 'inline']),
+  size?: 'small' | 'medium' | 'large'
+  layout?: 'stacked' | 'inline'
   /**
    * the textarea will expand vertically to fit the height of the content,
    * unless its content exceeds `maxHeight`
    */
-  autoGrow: PropTypes.bool,
+  autoGrow?: boolean
   /**
    * is the textarea resizable (in supported browsers)
    */
-  resize: PropTypes.oneOf(['none', 'both', 'horizontal', 'vertical']),
+  resize?: 'none' | 'both' | 'horizontal' | 'vertical'
   /**
    * a fixed width for the textarea
    */
-  width: PropTypes.string,
+  width?: string
   /**
    * Initial height for the textarea (if autoGrow is true it will grow vertically)
    * Accepts CSS units, e.g. '55px'
    */
-  height: PropTypes.string,
+  height?: string
   /**
    * when autoGrow is true, the textarea will never grow beyond this value
    */
-  maxHeight: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
+  maxHeight?: number | string
   /**
    * object with shape: `{
    * text: PropTypes.node,
    * type: PropTypes.oneOf(['error', 'hint', 'success', 'screenreader-only'])
    *   }`
    */
```

Comment:
Matyi said in one of my PR-s that this would be more readable as:

```suggestion
  /**
   * Array of objects with shape: `{
   *   text: React.ReactNode,
   *   type: One of: ['error', 'hint', 'success', 'screenreader-only']
   * }`
   */
```